tests/unit/test_bruteforce.cpp

```diff
@@ -871,3 +872,65 @@ TEST_F(BruteForceTest, brute_force_resolve_params) {
     VecSimIndex_Free(index);
     array_free(rparams);
 }
+
+TEST_F(BruteForceTest, brute_get_distance) {
+    size_t n = 4;
+    size_t dim = 2;
+    size_t numIndex = 3;
+    VecSimIndex *index[numIndex];
+    std::vector<double> distances;
+
+    float v1[] = {M_PI, M_PI};
+    float v2[] = {M_E, M_E};
+    float v3[] = {M_PI, M_E};
+    float v4[] = {M_SQRT2, -M_SQRT2};
+
+    VecSimParams params{
+        .algo = VecSimAlgo_BF,
+        .hnswParams = HNSWParams{.type = VecSimType_FLOAT32, .dim = dim, .initialCapacity = n}};
+
+    for (size_t i = 0; i < numIndex; i++) {
+        params.bfParams.metric = (VecSimMetric)i;
+        index[i] = VecSimIndex_New(&params);
+        VecSimIndex_AddVector(index[i], v1, 1);
+        VecSimIndex_AddVector(index[i], v2, 2);
+        VecSimIndex_AddVector(index[i], v3, 3);
+        VecSimIndex_AddVector(index[i], v4, 4);
+        ASSERT_EQ(VecSimIndex_IndexSize(index[i]), 4);
+    }
+
+    void *query = v1;
+    double dist;
+
+    // VecSimMetric_L2
+    distances = {0, 0.3583844006061554, 0.1791922003030777, 23.739208221435547};
```

Comment:
haven't we talked about calling the space itself?

Reply:
yes, but now we know that the results will be consistent across different hardware and optimizations.
I can change that but Ill have to externalize the metric function